### DISCUSSION TITLE
Template doesn't react to property change in store of array of objects

### DISCUSSION THREAD

**Discussion by stereokai** [solidjs/solid#2293] [Q&A]
Using either of the 2 methods, when I log the users store, I see that the name of the user has changed in the store object. However, the template doesn't update/react to the new name. Why is that?

```jsx
// user: { name: string }

const [users, setUsers] = createStore([{ name: "some user" }]);
const [userName, setUserName] = createSignal("");

currUser = users[0];
setUserName("newName")

// Store update, method 1:
setUsers((user) => user === currUser, "name", userName());
// Store update, method 2:
setUsers(
  produce((users) => {
    const user = users.find((user) => user === currUser);
    if (user) {
      user.name = userName();
    }
  }),
);

// JSX:
      <For each={users}>
        {(user, index) => {
          return (
            <div>
                {`${user.name}`}
            </div>
          );
        }}
      </For>

```

**Comment by atk** [comment]
As discussed on discord, stores are not reference-safe; the object accessed in the store is a Proxy with a different reference than the underlying object, so currUser can not ever be equal to user.

**Reply by stereokai** [reply]
Please explain this then? Am I missing something? They are matching (`Return value: true`)
<img width="881" alt="image" src="https://github.com/user-attachments/assets/9930f2fa-b207-4892-9724-cfbe9ddb70dd">
